quickannotator/api/v1/image/routes.py

```diff
@@ -97,52 +108,57 @@ def post(self, args):
-                os.makedirs(slide_folder_path, exist_ok=True)
-                shutil.move(temp_filepath, image_full_path)
-
-                new_image.path = image_full_path
-                db_session.add(new_image)
-                db_session.commit()
+                image_id = save_image_from_file(project_id, file)
 
+                # get all annotation class name
+                annotation_classes = get_all_annotation_classes()
                 # import annotation if it exist in temp dir
-                for format in AnnotationFileFormats:
-                    temp_image_path = fsmanager.nas_write.get_temp_image_path(relative=False)
-                    annot_filepath = os.path.join(temp_image_path, f'{file_basename}_annotations.{format.value}')
-                    # for geojson
-                    if os.path.exists(annot_filepath):
-                        import_annotations(image_id, annot_filepath)
+                for annot_cls in annotation_classes:
+                    for format in constants.AnnotationFileFormats:
+                        temp_image_path = fsmanager.nas_write.get_temp_image_path(relative=False)
+                        annot_filepath = os.path.join(temp_image_path, f'{file_basename}_{annot_cls.name}_annotations.{format.value}')
+                        # for geojson
+                        if os.path.exists(annot_filepath):
+                            # ?? should always ground true is True ??
+                            import_annotations(image_id, annot_cls.id , True, annot_filepath)
+                
             # handle annotation file
             if file_ext in JSON_extensions:
-                temp_image_path = fsmanager.nas_write.get_temp_image_path(relative=False)
-                annot_filepath = os.path.join(temp_image_path, filename)
+                save_annotation_file_to_temp_dir(file)
+
+            # handle tsv file
+            if file_ext in TABULAR_extensions:
+                # save tsv under current project folder
+
+                tsv_filepath = save_tsv_to_temp_dir(project_id, file)
+
+
+                # read tsv file
+                header = 0
+                col_name_filename = constants.TSVFields.FILE_NAME.value
+                if isHistoqcResult(tsv_filepath):
+                    header = constants.TSVFields.HISTO_TSV_HEADLINE.value - 1
+                    col_name_filename = constants.TSVFields.HISTO_FILE_NAME.value 
+                data = pd.read_csv(tsv_filepath, sep='\t', header=header, keep_default_na=False)
+                columns = data.columns
+
+                actor_name = compute_actor_name(project_id, constants.NamedRayActorType.ANNOTATION_EXPORTER)
+                importer = AnnotationImporter.options(name=actor_name).remote()
+                
+                # check if the actor is done
+                tester_ids = [importer.import_from_tsv_row.remote(project_id, col_name_filename, row, columns) for index, row in data.iterrows()]
```

Comment:
need to instantiate 1 ray actor (with a unique actor name) per row. Within the import_from_tsv_row, update the progress attribute of the ray actor